### PR TITLE
Dev - improvements to `processing.lazy_merge_arrays`

### PR DESCRIPTION
**Pull request by DakotaHester**
Fixes/improvements to `processing.lazy_merge_arrays, including default parameter handling and more robust to various `nodata` edge cases (e.g., `nodata is None`).  

Directly addresses:
- https://github.com/gcermsu/pcxarray/issues/4
- https://github.com/gcermsu/pcxarray/issues/5

Partially addresses:
- https://github.com/gcermsu/pcxarray/issues/7 (can't fully test to ensure that issue is reolved at the moment)

A future update should look at all the edge cases for nodata values in `lazy_merge_arrays` and write some tests.

### CODE REVIEW

File: src/pcxarray/processing.py [hunk 1]
```diff
@@ -250,7 +260,7 @@ def prepare_data(
             try:
                 da = read_single_item(
                     item_gs=item,
-                    geometry=geometry,
+                    geometry=geometry.buffer(1) if crs.is_projected else geometry, # buffer slightly to avoid edge cases
```
[the remark this answers is not in the document]
Comment (DakotaHester) [reply src/pcxarray/processing.py:263]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/gcermsu/pcxarray/pull/8#discussion_r2495561275)

File: src/pcxarray/processing.py [hunk 2]
```diff
@@ -305,7 +315,7 @@ def safe_read_item(item_series):
             try:
                 return read_single_item(
                     item_gs=item_series,
-                    geometry=geometry,
+                    geometry=geometry.buffer(1) if crs.is_projected else geometry,
```
[the remark this answers is not in the document]
Comment (DakotaHester) [reply src/pcxarray/processing.py:318]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/gcermsu/pcxarray/pull/8#discussion_r2495561291)
